### DISCUSSION TITLE
[React] Nested Dialog

### DISCUSSION THREAD

**Discussion by Marsony** [floating-ui/floating-ui#2228] [Q&A] [answered]
Hello everyone, is it possible to use nested Dialogs together with the FloatingPortal?

I got the Dialog component from the floating-ui docs (modified by https://github.com/floating-ui/floating-ui/discussions/2206), this component haven't implementation of the nesting.

I tried to use useDismiss({ bubbles: false }), but it doesn't help.

Thanks!

UPD: Right now when i opened nested dialog, after click on any area my first dialog closing.

**Comment by atomiks** [comment, marked as answer]
How are you using the component? It works fine seen here - https://codesandbox.io/s/hopeful-hoover-9kqb0x?file=/src/App.tsx

**Reply by Marsony** [reply]
> How are you using the component? It works fine seen here - https://codesandbox.io/s/hopeful-hoover-9kqb0x?file=/src/App.tsx

Yes, here's all works fine, ty)

But, I have a little different case.
For example, Im opening ?auth=register as Dialog, and together with the ?auth=register, Im opening another Dialog on another part of my application.

Case in your sandbox: Dialog inside another Dialog (a > b)
My case: Dialog with Dialog (first not inside the second). (a + b)

**Reply by atomiks** [reply]
Yeah in this case the React tree checks won't work. [The function guard option should help](https://floating-ui.com/docs/useDismiss#outsidepress)

**Reply by Marsony** [reply]
> Yeah in this case the React tree checks won't work. [The function guard option should help](https://floating-ui.com/docs/useDismiss#outsidepress)

Thank you for the answer!

Resolved by `e.target?.contains(data.elements?.floating)`

**Reply by Marsony** [reply]
> Yeah in this case the React tree checks won't work. [The function guard option should help](https://floating-ui.com/docs/useDismiss#outsidepress)

But idk what can I do with the "Esc" key. With that guard I can fix only `outsidePressEvent`

Maybe use react-context?

**Reply by atomiks** [reply]
Esc key is indeed a problem. I _think_ you'll need to use `FloatingTree` and [pass in a custom parent](https://floating-ui.com/docs/FloatingTree#custom-parent). Btw here's some discussion about this earlier:  https://github.com/floating-ui/floating-ui/issues/1982

**Reply by Marsony** [reply]
> Клавиша Esc действительно проблема. Я _думаю_ , вам нужно будет использовать `FloatingTree`и [передать пользовательский parent](https://floating-ui.com/docs/FloatingTree#custom-parent) . Кстати, вот обсуждение этого ранее: #1982

Every Dialog already wrapped in FloatingTree, but idk how can it help to resolve "Esc" button

**Reply by atomiks** [reply]
There must be only one FloatingTree wrapping both Dialogs. The Dialogs `useFloatingNodeId()` which subscribes them to the tree. Then it should work when marking the sibling as a child (`useFloatingNodeId(parentId)`) of the one you want as a parent.

**Reply by atomiks** [reply]
Demo: https://codesandbox.io/s/optimistic-cannon-rhwh81?file=/src/App.tsx

**Reply by Marsony** [reply]
> Demo: https://codesandbox.io/s/optimistic-cannon-rhwh81?file=/src/App.tsx

Seems like cool, but I fix this trouble with event-emitter (useFloatingTree)

Thanks!
